### PR TITLE
Multiple directories

### PR DESCRIPTION
**Pull request by axoTheLotl**
Hey There,

I created the possibility for users to add multiple base directories. Only the content of the checked directories are shown in the list view of the main activity. This makes it possible to have e.g. a music and audiobook directory and switch between them easily. Removing directories from the directories list will result in removal of all associated data (albums, audio_files, bookmarks) from the database.

Before solving the remaining issues, I wanted to check if you are interested in this functionality.

Remaining (known) issues are:
- duplicate Album names (in different directories) will result in errors
- update of the App won't account for data already in the database
- some minor display glitches (e.g. trash bin icon is not centered)

If you like the basic idea I'd be happy to implement solutions for these issues and any other necessary changes.    

Looking forward to your feedback.
best
Lotl

### CODE REVIEW

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -0,0 +1,164 @@
+<component name="ProjectCodeStyleConfiguration">
```
Comment (Sulter) [inline .idea/codeStyles/Project.xml]: this is a IDE-based config file, we should probably add it to .gitignore?

**Comment by flackbash** [comment]
I tried to run your code, but it does not compile. Some strings are missing in the strings.xml file. I added them locally so I could test your PR. But I got a NullPointerException when clicking on an album item in the MainActivity (in line 132) and the album covers were not displayed correctly.

While I think this functionality would be nice to have, I don't think there is a great need for it and I have not added it until now because it requires a lot of changes (as you probably noticed ;) ) and - depending on how well it is implemented - can make the whole app a lot more fragile and difficult to maintain. So out of the list of things that would be nice to have or should be fixed, this is definitely one of the harder ones to implement and quite an ambitious goal you have set for yourself as a beginner. A lot is working nicely (though I haven't had a close look at how you implemented it), but there also seem to be quite a few serious issues with it so far. I think the chances that I will merge it eventually are not too high. You can of course try ;) Otherwise you can find yourself a bug or feature that requires less changes to less files which will increase the chances of it being merged :)

In general, this article might be useful for writing good commit messages https://chris.beams.io/posts/git-commit/ .

File: .idea/codeStyles/Project.xml (hunk 1 above)
Comment (flackbash) [reply .idea/codeStyles/Project.xml]: Hm yes. We could probably add the entire codeStyles directory to .gitignore.

**Comment by axoTheLotl** [comment]
Thank you for your comments!

I realized that a whole lot of changes are necessary for a feature that is not so important after all and also that this is not a really beginner project...
Still learned a lot and wanted to give it a shot before putting this on ice.

Thanks also for the link on writing commit messages. I did not realize the mess of commits I created locally will be visible in the pull request : )

I'll keep to simpler features for now and work my way up.

**Comment by flackbash** [comment]
That sounds good :)
And you definitely came a long way with this one already, touching so many different topics... I'm sure you learned a lot.

**Pull request closed without merging**